### PR TITLE
Lower the number of objects stored in the heap old gen

### PR DESCRIPTION
**Pull request by gwenneg**
I use JFR / MissionControl for my `policies-history` work to check the impact of code changes on the prod memory leak (which is really easy to reproduce locally BTW). Heap old gen analysis showed a lot of `ArrayList`/`HashSet` resizes and also cache `read` operations after `put` operations while we don't use the returned values. This PR contains short-term optimizations that will lower the number of objects stored in the old gen. It only mitigates some of the consequences of the leak but does not fix the cause at all. Don't expect a fantastic result, but there will be a slight improvement.

See [this article](https://infinispan.org/blog/2012/09/04/speeding-up-cache-calls-with) about the reason why we should use `Flag.IGNORE_RETURN_VALUES`.

### CODE REVIEW

File: engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java [hunk 1]
```diff
@@ -138,7 +140,7 @@ public void addActionDefinition(String tenantId, ActionDefinition actionDefiniti
         if (found != null) {
             throw new FoundException(pk);
         }
-        backend.put(pk(actionDefinition), new IspnActionDefinition(actionDefinition));
+        backend.getAdvancedCache().withFlags(IGNORE_RETURN_VALUES).put(pk(actionDefinition), new IspnActionDefinition(actionDefinition));
```
Comment (pilhuhn) [inline engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java:143]: Would it make sense to turn `backend` into a new class `SuperCache` (that inherits from `Cache`) which then gets a method `putNR()`, which internally supplies the flag? This way you would not need the `.getAC().withFlags()` all over the place

Comment (gwenneg) [reply engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java:143]: If we wanted to keep using `backend`, yes definitely. Now, with policies-history coming soon (hopefully!) I don't know if that's worth it.

Comment (gwenneg) [reply engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java:143]: @pilhuhn Can we merge the current version or would you prefer the improved one?

Comment (pilhuhn) [reply engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java:143]: Whatever you find right :)

Comment (gwenneg) [reply engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java:143]: Let's merge this then, I'd rather stay focused on the main part 😃

**Review by josejulio — APPROVED** [review]

**Pull request merged**